core/src/main/java/com/linecorp/armeria/client/WebClientBuilder.java

```diff
@@ -183,6 +188,39 @@ public WebClientBuilder decorator(DecoratingHttpClientFunction decorator) {
         return (WebClientBuilder) super.decorator(decorator);
     }
 
+    /**
+     * Adds the decorator that decorates the header of request that name is specified {@code name}.
+     *
+     * @param name  The name of the header.
+     * @param value A supplier that provides a future of the value of the header.
+     */
+    public WebClientBuilder headerDecorator(CharSequence name,
+                                            Supplier<CompletableFuture<Object>> value) {
+        requireNonNull(name, "name");
+        requireNonNull(value, "value");
+        final DecoratingHttpClientFunction decorator = (client, ctx, req) -> {
+            final CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+            value.get().handle((header, cause) -> {
```

Comment:
Should the future that results from `value.get()` be converted into a context-aware future using `RequestContext#makeContextAware`?